src/cunumeric/index/wrap.cu

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/wrap.h"
+#include "cunumeric/index/wrap_template.inl"
+#include "cunumeric/cuda_help.h"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <int DIM>
+__global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
+  wrap_kernel(const AccessorWO<Point<DIM>, 1> out,
+              const int64_t start,
+              const int64_t volume,
+              const Pitches<0> pitches_out,
+              const Point<1> out_lo,
+              const Pitches<DIM - 1> pitches_in,
+              const Point<DIM> in_lo,
+              const size_t in_volume)
+{
+  const auto idx = global_tid_1d();
+  if (idx >= volume) return;
+  const int64_t input_idx = (idx + start) % in_volume;
+  auto p                  = pitches_in.unflatten(input_idx, in_lo);
+  out[idx]                = p;
```

Comment:
I think the old code was appropriate, because `out` here is globally indexed:
```
  const int64_t input_idx = (idx + start) % in_volume;
  auto out_p             = pitches_out.unflatten(idx, out_lo);
  auto p                 = pitches_in.unflatten(input_idx, in_lo);
  out[out_p]             = p;
```